File: A - SOURCE CODE/A - SERVICE/Shared/ViewModels/Categories/LoadCategoryViewModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;
using SystemConstant.Enumerations.Order;
using SystemConstant.Models;
using Shared.Models;

namespace Shared.ViewModels.Categories
{
    public class LoadCategoryViewModel
    {
        #region Properties

        /// <summary>
        ///     Id of Category.
        /// </summary>
        [Required]
        [MinLength(1)]
        public List<int> Ids { get; set; }

        /// <summary>
        /// Sorted property & direction.
        /// </summary>
        public Sort<CategoriesSort> Sort { get; set; }

        /// <summary>
        ///     Pagination of records filter.
        /// </summary>
        public Pagination Pagination { get; set; }

        #endregion
    }
}

File: A - SOURCE CODE/B - CLIENT/A - WEB/B - USER/src/app/view-models/load-category.view-model.ts
import {Pagination} from "../models/pagination";

export class LoadCategoryViewModel {

    //#region Properties

    // List of category group indexes.
    public categoryGroupIds: Array<number> | null;

    // Pagination.
    public pagination: Pagination | null;

    //#endregion

    //#region Constructor

    public constructor(){
        this.categoryGroupIds = null;
        this.pagination = new Pagination();
    }

    //#endregion
}